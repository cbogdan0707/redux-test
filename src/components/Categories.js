import React from "react";
import { connect } from "react-redux";

import { filterProductByCategory } from "../actions";

export const Categories = ({
  categories,
  activeCategory,
  filterProductByCategory,
  products
}) => {
  return categories.map(category => {
    const activeClass = `item ${
      activeCategory === category.id ? "active" : ""
    }`;
    return (
      <div
        className={activeClass}
        key={category.id}
        onClick={() => filterProductByCategory(products, category.id)}
      >
        {category.title}
      </div>
    );
  });
};

const mapStateToProps = state => {
  return {
    categories: state.categories,
    activeCategory: state.productsByCategory.categoryId,
    products: state.products
  };
};

export default connect(
  mapStateToProps,
  { filterProductByCategory }
)(Categories);
