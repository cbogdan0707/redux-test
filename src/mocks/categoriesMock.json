{
  "status": "ok",
  "meta": { "offset": 0, "limit": 1, "count": 1, "total": 1 },
  "data": [
    {
      "id": "faeedf8a-bf7d-11e5-a0f9-02fada0dd3b9",
      "title": "Drinks Cabinet"
    },
    {
      "id": "529ea59e-bf7e-11e5-840e-02fada0dd3b9",
      "title": "Kitchenware"
    },
    {
      "id": "fec10d0e-bf7d-11e5-90a9-02fada0dd3b9",
      "title": "Desserts"
    }
  ]
}
