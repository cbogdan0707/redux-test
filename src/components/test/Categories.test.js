import React from "react";
import renderer from "react-test-renderer";

import { Categories } from "../Categories";
import categoriesMock from "../../mocks/categoriesMock";

describe("Categories snapshot test", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Categories
          categories={categoriesMock.data}
          activeCategory={categoriesMock.data[2].id}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
