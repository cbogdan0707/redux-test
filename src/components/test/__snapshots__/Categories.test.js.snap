// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories snapshot test renders correctly 1`] = `
Array [
  <div
    className="item "
    onClick={[Function]}
  >
    Drinks Cabinet
  </div>,
  <div
    className="item "
    onClick={[Function]}
  >
    Kitchenware
  </div>,
  <div
    className="item active"
    onClick={[Function]}
  >
    Desserts
  </div>,
]
`;
